NuRadioReco/framework/base_station.py

```diff
@@ -76,20 +76,43 @@ def remove_parameter(self, key):
         self._parameters.pop(key, None)
 
     def set_station_time(self, time):
-        if time is None:
-            self._station_time = None
-            return
+        """ 
+        Set the (absolute) time for the station (stored as astropy.time.Time).
+        Not related to the event._event_time.
+        
+        Parameters
+        ----------
+        
+        time: astropy.time.Time or datetime.datetime or float
+            If time is a float it is interpreted as UTC unix timestamp.
+        """
+
         if isinstance(time, datetime.datetime):
-            time_strings = str(time).split(' ')
-            self._station_time = astropy.time.Time('{}T{}'.format(time_strings[0], time_strings[1]), format='isot')
+            self._station_time = astropy.time.Time(time)
+        elif isinstance(time, astropy.time):
+            self._station_time = time
+        elif time == None:
+            self._station_time = None
         else:
-            if time.format == 'datetime':
-                time_strings = str(time).split(' ')
-                self._station_time = astropy.time.Time('{}T{}'.format(time_strings[0], time_strings[1]), format='isot')
-            else:
-                self._station_time = time
+            # time is interpreted as unix utc timestamp
+            self._station_time = astropy.time.Time(time, format='unix')
```

Comment:
There are obviously a lot of float formats for time (e.g. 'mjd', ...). `astropy.time` therefore requires a format specification if time is given as a float. You do mention in the docstring that floats will be interpreted as unix by default, but even so, I would have a slight preference for copying the astropy convention (i.e. adding the argument `format=None` or `format='unix'`) to the `set_station_time` function, if you want to add this functionality.

Reply:
Have a look now.